### ISSUE TITLE
Weird const in the arrow functions demo

### ISSUE THREAD

**Issue by codepo8** [FormidableLabs/es6-interactive-guide#11]
You're defining the double in the last example as a const, however it gets changed on every application of the map. let would make more sense, wouldn't it?

**Comment by exogen** [comment]
It's not really getting changed though on each call since each call has its own scope within the function; the declarations don't "carry over" on each call. `eslint` would actually warn you of an unnecessary use of `let`.

**Comment by codepo8** [comment]
True, it does seem to be an odd use of const though as it is never re-used. Maybe a simple var should do the trick. It's not a mistake, but it seems a weird use, seeing that defining a const is a statement of "this won't change"

**Comment by exogen** [comment]
I'd personally change the whole function to `return number * number *  number;` myself, the separate declaration is indeed kinda confusing no matter how it's declared. But FWIW, `eslint` would warn of `var` as well since it recommends only `let` or `const`.

**Comment by codepo8** [comment]
I agree with that option. Much less contrived. Or to make it shorter return
Math.pow(number, 3)

**Comment by divmain** [comment]
`number * number * number` is definitely better.  However:

> seeing that defining a const is a statement of "this won't change"
It's a bit more nuanced than that.  `const` is really a statement of "this won't be reassigned within the context of this block, and specific to this invocation".  Thinking of a `const` in terms of "this won't change" can lead to incorrect assumptions.  For example, `const a = {}; a.val = true;` is perfectly valid - `a` is not reassigned in this case, but the object that `a` references can change.

In the arrow-function example, `const` should be preferred over `var` (even though it is definitely unnecessary) because `a`'s assignment doesn't change within the context of each invocation.  In fact, I would argue that there is no longer a need to use `var` anywhere, assuming an ES2015+ environment.
